app/scripts/models/account.js

```diff
@@ -790,13 +790,40 @@ define(function (require, exports, module) {
         .then((oAuthApps) => {
           oAuthApps.map((item) => {
             item.clientType = Constants.CLIENT_TYPE_OAUTH_APP;
+            item.isOAuthApp = true;
           });
 
           return oAuthApps;
         });
     },
 
     /**
+     * Fetch the account's Web Sessions and populate into the collection
+     *
+     * @returns {Promise} resolves when the action completes
+     */
+    fetchSessions () {
+      return this._fxaClient.sessions(this.get('sessionToken'))
+        .then((sessions) => {
+          sessions.map((item) => {
+            if (item.isDevice) {
+              item.clientType = Constants.CLIENT_TYPE_DEVICE;
+              // override the item id as deviceId for consistency
+              // if you ever need the tokenId just add it here with a different name
+              item.id = item.deviceId;
+            } else {
+              item.clientType = Constants.CLIENT_TYPE_WEB_SESSION;
+              item.name = 'Web Session';
```

Comment:
Does `Web Session` need to be translated? 

It almost seems like this transmogrification of the items could be done in the fxa-client since the data being modified is that being returned by the backend, and the data isn't being used by the account directly.